students/luke/s03/mailroom.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+"""
+https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom.html#exercise-mailroom
+"""
+
+class donor:
+    """Donor class holds donor name and list of donations"""
+    def __init__(self, name, donations=None):
+        self.name = name
+        if donations is None:
+            self.donations = [ ]
+        else:
+            self.donations = donations
+
+
+def driver():
+    """ Main loop for mailroom """
+    done = False
+    while not done:
+        action = input(("Select an action and press return:\n"
+            "[S]end a thank-you (add users and donation entries)\n"
+            "[C]reate a report\n"
+            "[Q]uit\n"
+            "> "))
+        action = action.lower()
+        while not action in "scq": action = input("Bad option, reenter > ")
```

Comment:
You can also do this in one line: 

            while not action.lower() in "scq": action = input("Bad option, reenter > ")
And you can use the same concept to automatically coerce the user input to lowercase if you want.

Reply:
Fixed in next revision of s04/mailroom.py (not this one).  This was just a sloppy iteration of my input validation.